target/ppc/translate/storage-ctrl-impl.c.inc

```diff
@@ -0,0 +1,93 @@
+/*
+ * Power ISA decode for Storage Control instructions
+ *
+ * Copyright (c) 2022 Instituto de Pesquisas Eldorado (eldorado.org.br)
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, see <http://www.gnu.org/licenses/>.
+ */
+
+/*
+ * Store Control Instructions
+ */
+
+static bool do_tlbie(DisasContext *ctx, arg_X_tlbie *a, bool local)
+{
+#if defined(CONFIG_USER_ONLY)
+    GEN_PRIV2(true);
+#else
+    TCGv_i32 t1;
+    int rb, ric;
+    bool prs, r;
+
+    rb = a->rb;
+    ric = a->ric;
+    prs = a->prs;
+    r = a->r;
+
+    if (ric || prs || r) {
+        REQUIRE_INSNS_FLAGS2(ctx, ISA300);
+    }
```

Comment:
As we saw in our POWER8 and POWER9, before Power ISA 3.0 these bits were ignored, so we should have something like:
```suggestion
    if ((ctx->insns_flags2 & PPC2_ISA300) == 0) {
        /* Before Power ISA 3.0, the corresponding bits of RIC, PRS, and R
         * were a reserved field and should be ignored. */
        ric = 0;
        prs = 0;
        r = 0;
    }
```
It might worth adding a comment in the email for this patch (after the `---` but before the patch itself) pointing to the section 1.3.3 of Power ISA that differentiates reserved fields and reserved values.